Always inline Index methods.

korran opened the pull request:
This significantly improves code-gen for many use cases.

korran committed:
Always inline Index methods.

eldruin reviewed (commented):
Sounds good to me.
Could you add an entry to the changelog?

korran committed:
Support unsized types in std trait delegations.

This makes it possible to use PartialEq, Debug, etc with unsized types
such as `Aligned<A4, [u8]>`.
